urwid/event_loop/asyncio_loop.py

```diff
@@ -67,20 +66,37 @@ class AsyncioEventLoop(EventLoop):
     def __init__(self, *, loop: asyncio.AbstractEventLoop | None = None, **kwargs) -> None:
         super().__init__()
         self.logger = logging.getLogger(__name__).getChild(self.__class__.__name__)
-        if loop:
-            self._loop: asyncio.AbstractEventLoop = loop
+
+        if sys.version_info[:2] < (3, 11):
             self._event_loop_policy_altered: bool = False
             self._original_event_loop_policy: asyncio.AbstractEventLoopPolicy | None = None
-        else:
-            self._original_event_loop_policy = asyncio.get_event_loop_policy()
-            if IS_WINDOWS and not isinstance(self._original_event_loop_policy, asyncio.WindowsSelectorEventLoopPolicy):
-                self.logger.debug("Set WindowsSelectorEventLoopPolicy as asyncio event loop policy")
-                asyncio.set_event_loop_policy(asyncio.WindowsSelectorEventLoopPolicy())
-                self._event_loop_policy_altered = True
+
+            if loop:
+                self._loop: asyncio.AbstractEventLoop = loop
             else:
-                self._event_loop_policy_altered = False
+                self._original_event_loop_policy = asyncio.get_event_loop_policy()
+                if sys.platform == "win32" and not isinstance(
+                    self._original_event_loop_policy, asyncio.WindowsSelectorEventLoopPolicy
+                ):
+                    self.logger.debug("Set WindowsSelectorEventLoopPolicy as asyncio event loop policy")
+                    asyncio.set_event_loop_policy(asyncio.WindowsSelectorEventLoopPolicy())
+                    self._event_loop_policy_altered = True
+                else:
+                    self._event_loop_policy_altered = False
+
+                self._loop = asyncio.get_event_loop()
 
-            self._loop = asyncio.get_event_loop()
+        else:
+            self._runner: asyncio.Runner | None = None
+
+            if loop:
+                self._loop: asyncio.AbstractEventLoop = loop
+            else:
+                try:
+                    self._loop = asyncio.get_running_loop()
```

Comment:
https://docs.python.org/release/3.12.10/library/asyncio-eventloop.html#asyncio.get_running_loop

No running loop -> we can use runner and and offload everything to the new asyncio API standard